### PR TITLE
fix(participant): emit close if the websocket doesn't exist

### PR DESCRIPTION
**Pull request by DeekyJay**

### CODE REVIEW

File: src/participant.ts [hunk 1]
```diff
@@ -530,6 +530,12 @@ export class Participant extends EventEmitter {
   private handleWebsocketError(ev: Event) {
     // tslint:disable-next-line
     if (!this.websocket || !this.websocket.url) {
+      this.emit('close', {
+        code: -1,
```
Comment (connor4312) [inline src/participant.ts]: Should probably be `1006`.

> sed to indicate that a connection was closed abnormally (that is, with no close frame being sent) when a status code is expected.

**Comment by connor4312** [comment]
I'm not totally clear on the scenario/use case of this, can you describe a bit more?

File: src/participant.ts [hunk 2]
```diff
@@ -530,6 +530,12 @@ export class Participant extends EventEmitter {
...
+        message: 'No websocket or websocket url',
+        expected: this.state === State.Closing,
+        ev,
+      });
```
[the remark this answers is not in the document]
Comment (rogarMS) [reply src/participant.ts]: So the error we had on mobile was after disconnect was 'undefined is not an object (evaluating 'this.websock.url.replace') (On version 0.2.16)

0.2.17 added the if check, but now won't emit close, so we never know when to reconnect.

Maybe could be more granular, no websocket = no emit, websocket but no url = emit.

**Comment by rogarMS** [comment]
@connor4312 
On mobile, websocket doesn't have a url field.
So  after disconnect we were getting an error of 'undefined is not an object (evaluating 'this.websock.url.replace') (On version 0.2.16)

0.2.17 added the if check, but now won't emit close, so we never know when to reconnect.

**Comment by connor4312** [comment]
Okay, that makes sense, can that explanation be --amend'ed into the commit message for future people?

**Comment by connor4312** [comment]
Actually we should be able to remove the below fetch code now, since Constellation now always will upgrade the request into a socket so that it can send a close code down for clients (like the js) who don't make failure status codes available.

File: src/participant.ts [hunk 3]
```diff
@@ -525,39 +525,20 @@ export class Participant extends EventEmitter {
   /**
    * handleWebsocketError is called when the websocket emits an `error`. This
    * is generally called when the connection is terminated before a socket
-   * connection is established. We want to go back and get the error code/body.
+   * connection is established.
    */
   private handleWebsocketError(ev: Event) {
     // tslint:disable-next-line
     if (!this.websocket || !this.websocket.url) {
```
Comment (connor4312) [inline src/participant.ts]: Also remove this if statement, always run this code

**Pull request merged**